src/textual/_resolve.py

```diff
@@ -1,9 +1,10 @@
 from __future__ import annotations
 
-import sys
 from fractions import Fraction
 from itertools import accumulate
-from typing import cast, Sequence, TYPE_CHECKING
+from typing import Sequence, cast, TYPE_CHECKING
+
+from typing_extensions import Literal
```

Comment:
In this PR, I've seen that sometimes you put the "`from typing_extensions import ...`" inside an `if TYPE_CHECKING` and sometimes you don't. What determines which one to go for?

Reply:
It's not likely intentional, and more likely a side effect of mass editing files. There's not much difference either way, unless the object is used at runtime, and then it can't go inside a `TYPECHECKING` conditional.